File: app/src/main/java/fbla/game/Door.java
package fbla.game;

import com.google.gson.annotations.SerializedName;

public class Door {
    @SerializedName("x")
    private int x;

    @SerializedName("y")
    private int y;

    @SerializedName("target_level")
    private int targetLevel;

    @SerializedName("target_x")
    private int targetX;

    @SerializedName("target_y")
    private int targetY;

    @SerializedName("image")
    private String imagePath;

    private transient int textureId;

    public Door(int x, int y, int targetX, int targetY, int targetLevel, String imagePath){
        this.x = x;
        this.y = y;
        this.targetX = targetX;
        this.targetY = targetY;
        this.imagePath = imagePath;
        this.targetLevel = targetLevel;
    }

    // Getters:
    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public int getTargetLevel() {
        return targetLevel;
    }

    public int getTargetX() {
        return targetX;
    }

    public int getTargetY() {
        return targetY;
    }

    public void setTextureId(int id){
        this.textureId = id;
    }
    
    public int getTextureId(){
        return textureId;
    }

    public String getImagePath(){
        return imagePath;
    }

}

File: editor/html-rewrite/js/Door.js
class Door {
    x;
    y;
    target_level;
    target_x;
    target_y;
    image;
    constructor (x, y, target_level, target_x, target_y, image){
        this.x = x;
        this.y = y;
        this.target_level = target_level;
        this.target_x = target_x;
        this.target_y = target_y;
        this.image = image;
    }
    get getX(){
        return this.x;
    }
    get getY(){
        return this.y;
    }
    get getTargetLevel(){
        return this.target_level;
    }
    get getTargetX(){
        return this.target_x;
    }
    get getTargetY(){
        return this.target_y;
    }
    get getImage(){
        return this.image;
    }
}